crates/rpc/rpc/src/eth/api/traits/state.rs

```diff
@@ -0,0 +1,173 @@
+//! Loads state from database. Helper trait for `eth_`transaction and state RPC methods.
+
+use futures::Future;
+use reth_primitives::{Address, BlockId, BlockNumberOrTag, Bytes, B256, U256};
+use reth_provider::{StateProviderBox, StateProviderFactory};
+use reth_rpc_types::{serde_helpers::JsonStorageKey, EIP1186AccountProofResponse};
+use reth_rpc_types_compat::proof::from_primitive_account_proof;
+use reth_transaction_pool::{PoolTransaction, TransactionPool};
+
+use crate::{
+    eth::{
+        api::SpawnBlocking,
+        error::{EthApiError, EthResult, RpcInvalidTransactionError},
+    },
+    EthApiSpec,
+};
+
+/// Helper methods for `eth_` methods relating to state (accounts).
+pub trait EthState: LoadState + SpawnBlocking {
+    /// Returns a handle for reading data from transaction pool.
+    ///
+    /// Data access in default trait method implementations.
+    fn pool(&self) -> &impl TransactionPool;
```

Comment:
we could auto impl for & as well and remove the & here perhaps?

https://github.com/paradigmxyz/reth/blob/c7338593aa898b1cc82ca5a467997f6962139305/crates/transaction-pool/src/traits.rs#L42-L42